### PR TITLE
Adjust local planner to support arbitrary horizons and match length of body and foot plans

### PR DESCRIPTION
**Pull request by jcnorby**
Changes include
- Increase standard horizon length from 24 to 25
- Replace most instances of N+1 with N and N with N-1
- Modify yaml files and tests to match this

### CODE REVIEW

File: local_planner/src/local_planner.cpp [hunk 1]
```diff
@@ -75,16 +75,16 @@ LocalPlanner::LocalPlanner(ros::NodeHandle nh)
...
   foot_positions_world_ = Eigen::MatrixXd::Zero(N_, num_feet_ * 3);
   foot_velocities_world_ = Eigen::MatrixXd::Zero(N_, num_feet_ * 3);
   foot_accelerations_world_ = Eigen::MatrixXd::Zero(N_, num_feet_ * 3);
   foot_positions_body_ = Eigen::MatrixXd::Zero(N_, num_feet_ * 3);
   current_foot_positions_body_ = Eigen::VectorXd::Zero(num_feet_ * 3);
   current_foot_positions_world_ = Eigen::VectorXd::Zero(num_feet_ * 3);
   current_foot_velocities_world_ = Eigen::VectorXd::Zero(num_feet_ * 3);
-  ref_ground_height_ = Eigen::VectorXd::Zero(N_ + 1);
-  grf_plan_ = Eigen::MatrixXd::Zero(N_, 12);
+  ref_ground_height_ = Eigen::VectorXd::Zero(N_);
```
Comment (ardalantj) [inline local_planner/src/local_planner.cpp:86]: Maybe a comment clarifying the indexing

Comment (jcnorby) [reply local_planner/src/local_planner.cpp:87]: Updated.

**Comment by yanhaoy** [comment]
Is the size of foot_position intentionally using N_ instead of N_-1 for the adaptive complexity?

**Review by yanhaoy — APPROVED** [review]

**Comment by jcnorby** [comment]
Yes, this way there is a foot state for each body state (which adaptive complexity needs).

**Review by ardalantj — APPROVED** [review]

**Pull request merged**